src/NzbDrone.Common/Disk/DiskProviderBase.cs

```diff
@@ -293,18 +293,30 @@ public void SetPermissions(string filename, WellKnownSidType accountSid, FileSys
                 var sid = new SecurityIdentifier(accountSid, null);
 
                 var directoryInfo = new DirectoryInfo(filename);
-                var directorySecurity = directoryInfo.GetAccessControl();
+                var directorySecurity = directoryInfo.GetAccessControl(AccessControlSections.Access);
+
+                var rules = directorySecurity.GetAccessRules(true, false, typeof(SecurityIdentifier));
+
+                if (rules.OfType<FileSystemAccessRule>().Any(acl => acl.AccessControlType == controlType && (acl.FileSystemRights & rights) == rights && acl.IdentityReference.Equals(sid)))
+                {
+                    return;
+                }
 
                 var accessRule = new FileSystemAccessRule(sid, rights,
                                                           InheritanceFlags.ContainerInherit | InheritanceFlags.ObjectInherit,
-                                                          PropagationFlags.None, controlType);
+                                                          PropagationFlags.InheritOnly, controlType);
 
-                directorySecurity.AddAccessRule(accessRule);
-                directoryInfo.SetAccessControl(directorySecurity);
+                bool modified;
+                directorySecurity.ModifyAccessRule(AccessControlModification.Add, accessRule, out modified);
+
+                if (modified)
+                {
+                    directoryInfo.SetAccessControl(directorySecurity);
+                }
             }
             catch (Exception e)
             {
-                Logger.Warn(e, string.Format("Couldn't set permission for {0}. account:{1} rights:{2} accessControlType:{3}", filename, accountSid, rights, controlType));
+                Logger.Warn(e, "Couldn't set permission for {0}. account:{1} rights:{2} accessControlType:{3}", filename, accountSid, rights, controlType);
```

Comment:
1. No, it changed from string.Format to the appropriate override. It's just one I forgot when migrating to nlog 4.2. (Note that for Debug/Trace this Nlog override is faster, coz the format only executes when needed.
2. Avoid string interpolation for logging, coz it would execute before nlog is called. Roslyn has a ridiculous design flaw that makes it prefer casting it to string instead of using a FormattedString overload. (So if you have Warn(string) & Warn(FormattedString), then roslyn would cast the string interpolation (which is a FormattedString instance) to string and use Warn(string))